src/ducks/saksopplysninger.js

```diff
@@ -35,7 +36,47 @@ export function hentSaksopplysninger(fnr) {
     PENDING
   });
 }
+// selector
+const getPerson = (state) => state.saksopplysninger.data.person;
+const getAlleOrganisasjoner = (state) => state.saksopplysninger.data.organisasjoner;
+const getAlleArbeidsforhold = (state) => state.saksopplysninger.data.arbeidsforhold;
+const getArbeidsforholdet = (state, arbeidsforholdID) => {
+  if (state.saksopplysninger.status === 'OK') {
+    return state.saksopplysninger.data.arbeidsforhold.find((item) => item.arbeidsforholdIDnav.toString() === arbeidsforholdID);
+  }
+  return state;
+};
+const getOrganisasjon =  (state, orgnummer) => {
+  if (state.saksopplysninger.status === 'OK') {
+    return state.saksopplysninger.data.organisasjon.find((item) => item.orgnummer === orgnummer);
+  }
+  return state;
+};
+
+// reselect function
+export const getPersonState = createSelector(
+  [ getPerson ],
+  (person) => person
+);
+
+export const getAlleOrganisasjonerState = createSelector(
+  [ getAlleOrganisasjoner ],
+  (organisasjoner) => organisasjoner
+);
 
+export const getAlleArbeidsforholdState = createSelector(
+  [ getAlleArbeidsforhold ],
+  (arbeidsforhold) => arbeidsforhold
+);
+export const makeGetArbeidsforholdetState = () => createSelector(
+  getArbeidsforholdet,
```

Comment:
Om jeg forstår meg på reselect og dokumentasjonen der så bør man bytte om hvor logikken ligger. E.g
```
createSelector(
  (state, arbeidsforholdID) => arbeidsforholdID,
  (state) => state.saksopplysninger.data.arbeidsforhold || [],
  (arbeidsforholdID, arbeidsforhold) => arbeidsforhold.find((item) => item.arbeidsforholdIDnav === arbeidsforholdID)
);
```

Sånn jeg forstår det så kjøres alle funksjonene unntatt den siste ved alle stateendringer. Så man ønsker at de skal være så enkle som overhode mulig, og bare plukke ut biter av staten som man trenger for å gjøre selve logikken. Siste funksjonen er memoisert basert på outputen til de første funksjonene, så den vil da kun bli kjørt om det har skjedd en endring som er relevant for denne selectoren.

Reply:
Har jobbet videre med selectors og benyttet syntax som foreslått.